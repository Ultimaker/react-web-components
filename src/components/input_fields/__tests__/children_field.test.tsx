// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import ChildrenField, { ChildrenFieldProps } from '../children_field';

describe('The children field component', () => {
    let children;
    let props: ChildrenFieldProps;
    let wrapper;

    beforeEach(() => {
        children = <div>Children</div>;
        props = {
            id: 'testInputField',
            labelLayoutWidth: '1/1',
            labelWidthBreakpoint: 'sm',
            staticField: false,
            children,
        };
        wrapper = shallow(<ChildrenField {...props} />);
    });

    it('should render a wrapper', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.prop('children')).toEqual(children);
    });
});
