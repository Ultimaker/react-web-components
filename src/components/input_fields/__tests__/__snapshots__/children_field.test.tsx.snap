// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The children field component should render a wrapper 1`] = `
<InputFieldWrapper
  id="testInputField"
  labelLayoutWidth="1/1"
  labelWidthBreakpoint="sm"
  staticField={false}
>
  <div>
    Children
  </div>
</InputFieldWrapper>
`;
